vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java

```diff
@@ -125,4 +126,55 @@ public CreationReturnValue createImageBlock(String slideId, byte[] image, String
         return returnValue;
     }
 
+    @Override
+    public void updateTextBlock(TextBlock textBlock) {
+        textBlockRepo.save((TextBlock) textBlock);
+    }
+
+    @Override
+    public void updateImageBlock(IImageBlock imageBlock, byte[] image, String filename, Integer contentOrder)
+            throws ImageCouldNotBeStoredException {
+        IVSImage oldImage = imageBlock.getImage();
+        IVSImage slideContentImage = null;
+
+        if (image != null && image.length > 0) {
+            Tika tika = new Tika();
+            String contentType = tika.detect(image);
+            slideContentImage = imageFactory.createImage(filename, contentType);
+            slideContentImage = imageRepo.save((VSImage) slideContentImage);
+            imageRepo.delete((VSImage) oldImage);
+        }
+
+        if (slideContentImage != null) {
+            String relativePath = null;
+            try {
+                relativePath = storage.storeFile(image, filename, slideContentImage.getId());
+            } catch (FileStorageException e) {
+                throw new ImageCouldNotBeStoredException(e);
+            }
+            slideContentImage.setParentPath(relativePath);
+            imageRepo.save((VSImage) slideContentImage);
+        }
+
+        imageBlock.setImage(slideContentImage);
+        imageBlockRepo.save((ImageBlock) imageBlock);
+    }
```

Comment:
a lot of this looks like the code for creating an image block; let's put that into methods and get rid of the code duplication.